### ISSUE TITLE
Cannot load_model in tensorflow backend

### ISSUE THREAD

**Issue by tonywu95** [fchollet/keras#3377]
I tried training the following code on github:

https://github.com/osh/KerasGAN/blob/master/mnist_gan.py

using tensorflow backend (although the original code uses theano backend, but nothing in particular depends on theano). Training is working well: training loss and generated images are all good, and I saved the generator after training:

`
generator.save("test.h5")
`

But when I try to reload the model, for example in ipython, I got the following error:

<img width="816" alt="exception" src="https://cloud.githubusercontent.com/assets/12599121/17316875/cb0299e0-5845-11e6-8f04-b624a41c9e96.png">

Any ideas what's going wrong?

**Comment by bernardohenz** [comment]
I am working in Theano backend, but for save/load I use as follow:


```python
#SAVING
# serialize model to JSON
model_json = model.to_json()
with open("model.json", "w") as json_file:
    json_file.write(model_json)
# serialize weights to HDF5
model.save_weights("model.h5")
print("Saved model to disk")
```

and for loading

```python
# load json and create model
json_file = open('model.json', 'r')
loaded_model_json = json_file.read()
json_file.close()
loaded_model = model_from_json(loaded_model_json)
# load weights into new model
loaded_model.load_weights("model.h5")
print("Loaded model from disk")
```

In this way you are saving/loading the architecture (on json) AND the weights (on h5)

**Comment by OOMMYY** [comment]
if you are load model in the webserver,you must create a session:
this may help you .
http://stackoverflow.com/questions/38575392/importing-keras-inside-of-django-causes-it-to-crash/43341182#43341182
import tensorflow as tf
with tf.Session():
 //my code to call the Keras unit
